src/mon/HealthMonitor.cc

```diff
@@ -803,9 +803,14 @@ void HealthMonitor::check_for_mon_down(health_check_map_t *checks)
 {
   int max = mon.monmap->size();
   int actual = mon.get_quorum().size();
-  const auto now = ceph::real_clock::now();
+  const auto rcnow = ceph::real_clock::now();
+  const auto created = mon.monmap->created.to_real_time();
+  const auto mcnow = ceph::coarse_mono_clock::now();
+  const auto starttime = mon.get_starttime();
+
   if (actual < max &&
-      now > mon.monmap->created.to_real_time() + g_conf().get_val<std::chrono::seconds>("mon_down_mkfs_grace")) {
+      (rcnow - created) > g_conf().get_val<std::chrono::seconds>("mon_down_mkfs_grace") &&
```

Comment:
Having been bit by implicit signed vs unsigned conversions of the time spans, I prefer doing the time math as it was before this edit, namely to compare `now > (start_time + grace)`

Reply:
I think the main concern would be `rcnow` being in the future (due to system clock changes) where the result of `rcnow - created` would be a negative timepoint. I think this would cause the condition to be false in both cases, no?

Reply:
Ack